mobify/spline
# Update spline

Add some missing mixins from [platform-scaffold](https://github.com/mobify/platform-scaffold). First step of removing `!important` style in platform-scaffold to work with new AMP product. 

## Changes Mixins
- `border` mixin: now use `$border-color` variable to allow to change border colours. 
- `text-normal` mixin: small tweak to use `base-text-weight` mixin.
- `text-quiet` mixin: small tweak to use `$font-size` value rather than `0.785em`.
- `text-bold` mixin: small tweak to use `base-text-weight` mixin. 

## New Mixins
- background color mixins: `bg-color-neutral-00`, `bg-color-neutral-10`,  `bg-color-neutral-15`, `bg-color-neutral-20`, `bg-color-neutral-40`, `bg-color-neutral-50`, `bg-color-neutral-60`, `bg-color-neutral-70`, `bg-color-brand`, `bg-color-accent`, `bg-color-primary-action`, `bg-color-secondary-action`, and `bg-color-success`.

- box shadow mixin: `box-shadow`.

- color mixins: `color-neutral-00`, `color-neutral-10`, `color-neutral-15`, `color-neutral-20`, `color-primary-action`, `color-secondary-action`, and `color-sale`.

- display mixins: `display`, `display-inline-block`, `display-inline`, and `display-none`. 

- flexbox mixins: `direction-column`, `direction-column-reverse`, `direction-row`, `direction-row-reverse`, and `flex-none`.

- heading mixins: `heading-level-1`, `heading-level-2`, `heading-level-3`, `heading-level-4`, `heading-level-5`, and `heading-level-6`.

- layout mixins: `position-relative`, `position-absolute`, `position-fixed`, `position-top`, `position-right`, `position-bottom`, `position-left`, `position-full`, and `base-bleed`.

- text mixins: `base-font-family`, `text-font-family`, `text-header-font-family`, `text-loaded-header-font-family`, `base-text-weight`, `text-thin`, `text-extra-light`, `text-light`, `text-regular`, `text-medium`, `text-semi-bold`, `base-text-size`, `text-size-smaller`, `text-size-small`, `text-size-normal`, `text-size-big`, `text-size-bigger`, `text-size-huge`, `line-height`, `letter-spacing`, and `text-uppercase`.

- z-index mixin: `z-index`.

## Breaking Changes
- depreciate: `bg-color-light-error`.
- depreciate: `row-reverse`, renamed to `direction-row-reverse`.
- depreciate: `position-context`, renamed to `position-relative`.
- depreciate: `text-all-caps`, renamed to `text-uppercase `.

## How to test this PR in Platform Scaffold
- Go to https://github.com/mobify/progressive-web-sdk/pull/603 (working in progress)
- `npm update` to update `node_modules` or `npm install` if you have not installed the `node_modules`.

then

- Go to https://github.com/mobify/platform-scaffold/pull/550 (working in progress)
- `npm update` to update `node_modules` or `npm install` if you have not installed the `node_modules`.
- `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Make sure there is no error. 

## TODOs:
- [ ] +1 (@jeffkamo, @yourpalsonja, @gsaray)

Reminder: [Be excellent to each other](https://github.com/mobify/developer-values#be-excellent-to-each-other). Follow [these guidelines](https://www.djangoproject.com/conduct/) to help you do that when giving feedback!
